packages/widgets/widget-utils/src/bootstrap/bootstrapWidgetStates.js

```diff
@@ -0,0 +1,43 @@
+import {
+  ATTRIBUTE_WIDGET_REF,
+  ATTRIBUTE_WIDGET_STATES,
+  ID_INITIAL_WIDGET_STATE_STYLES,
+  ID_WIDGET_STATE_STYLES
+} from './constants'
+import * as remoteLogger from './remoteLogger'
+
+const createStyleElement = id => {
+  const style = document.createElement('style')
+  style.id = id
+  document.head.appendChild(style)
+  return style
+}
+
+export const setGlobalStyles = (backendUrl, id, css) => {
+  try {
+    const style = document.getElementById(id) || createStyleElement(id)
+    style.innerHTML = css
+  } catch (error) {
+    remoteLogger.logException(backendUrl, `Could not apply global styles.`, error)
+  }
+}
+
+export const setWidgetStateStyles = (backendUrl, widgetKey, widgetStates) => {
+  const selector = widgetStates
+    .map(state => `[${ATTRIBUTE_WIDGET_REF}="${widgetKey}"][${ATTRIBUTE_WIDGET_STATES}~="${state}"]`)
```

Comment:
praise: I wasn't aware of the difference between `~=` and `*=`. thx for teaching me this and kudos for choosing this one 😄